src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8216,6 +8228,9 @@ void checkWellKnown(WellKnownMember member)
                             original.Parameters[1] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
                         {
                             patternSymbol = method;
+                            method.AddUseSiteInfo(ref useSiteInfo);
+                            diagnostics.Add(syntax, useSiteInfo);
+                            ReportDiagnosticsIfObsolete(diagnostics, method, syntax, hasBaseReceiver: false);
```

Comment:
>ReportDiagnosticsIfObsolete(diagnostics, method, syntax, hasBaseReceiver: false); [](http://example.com/codeflow?start=28&length=81)

I am very concerned about the amount of code duplication and future maintenance associated with code touched by this PR. Let’s discuss that offline.